Allow aligning the mag to any rotation

fiam opened the pull request:
Introduce 3 new variables which allow setting the decidegrees
for the mag sensor alignment. When any of these 3 variables
are non-zero, mag is assumed to be mounted off-board and
"align_mag" as well as the board alignment are ignored.

Settings are named align_mag_roll, align_mag_pitch and
align_mag_yaw.

Fixes #86
Fixes #1029

fiam committed:
Allow aligning the mag to any rotation

Introduce 3 new variables which allow setting the decidegrees
for the mag sensor alignment. When any of these 3 variables
are non-zero, mag is assumed to be mounted off-board and
"align_mag" as well as the board alignment are ignored.

Settings are named align_mag_roll, align_mag_pitch and
align_mag_yaw.

Fixes #86
Fixes #1029

digitalentity commented on src/main/sensors/compass.c:
```diff
@@ -368,8 +371,34 @@ void compassUpdate(timeUs_t currentTimeUs)
+             .angles.yaw = DECIDEGREES_TO_RADIANS(compassConfig()->yawDeciDegrees),
+         };
+
+         rotateV(&v, &r);
```
`rotateV` is quite expensive because it builds rotation matrix every time.

fiam committed:
Precalculate external mag rotation matrix

Introduce rotateByMatrix(), which rotates a vector by the given
matrix. Initialize the matrix on mag initialization, so it's not
recalculated on every mag reading.
